NETCORE/test/Microsoft.ApplicationInsights.AspNetCore.Tests/Extensions/ApplicationInsightsExtensionsTests/AddApplicationInsightsTelemetryTests.cs

```diff
@@ -37,6 +37,25 @@ namespace Microsoft.Extensions.DependencyInjection.Test
 
     public class AddApplicationInsightsTelemetryTests : BaseTestClass
     {
+        [Fact]
+        public static void TelemetryModuleResolvableWhenKeyedServiceRegistered()
```

Comment:
This test appears to pass even without your change.
I copied it locally and ran it in main branch.

Reply:
Interesting. It looks like without the fix, worker service can't find any registered `ITelemetryHttpModule` and the test fails. AspNetCore though finds `RequestTrackingTelemetryModule` registered so the test passes.

With the fix in place there are actually 7 `ITelemetryHttpModule`s registered. I updated the tests to validate all 7 are found when keyed service is added to the mix.

Reply:
Just pushed updates to the test to improve them further. Added validation that a) `DiagnosticsTelemetryModule` is present in the final list and b) if the user supplied one we don't end up with 2.